app/scripts/lib/rpc-method-middleware/handlers/ethereum-chain-utils.js

```diff
@@ -1,31 +1,38 @@
 import { errorCodes, rpcErrors } from '@metamask/rpc-errors';
+import {
+  Caip25CaveatType,
+  Caip25EndowmentPermissionName,
+  getPermittedEthChainIds,
+} from '@metamask/multichain';
 import {
   isPrefixedFormattedHexString,
   isSafeChainId,
 } from '../../../../../shared/modules/network.utils';
-import { CaveatTypes } from '../../../../../shared/constants/permissions';
 import { UNKNOWN_TICKER_SYMBOL } from '../../../../../shared/constants/app';
-import { PermissionNames } from '../../../controllers/permissions';
 import { getValidUrl } from '../../util';
 
 export function validateChainId(chainId) {
-  const _chainId = typeof chainId === 'string' && chainId.toLowerCase();
-  if (!isPrefixedFormattedHexString(_chainId)) {
+  const lowercasedChainId =
+    typeof chainId === 'string' ? chainId.toLowerCase() : null;
+  if (
+    lowercasedChainId !== null &&
```

Comment:
```suggestion
    lowercasedChainId === null ||
```

Is this meant to be like this? Otherwise parsing null may blow up later?

Reply:
this lowercasedChainId check isn't actually needed and `isPrefixedFormattedHexString()` fails as intended. Removing this extra check here

Reply:
https://github.com/MetaMask/metamask-extension/pull/27847/commits/2657bfefee336b73b75f0e1989e13b52ab75159d

Reply:
thanks for catching this